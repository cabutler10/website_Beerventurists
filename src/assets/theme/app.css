html {
  height: 100%;
}

body {
  margin: 0;
  padding: 0;
  font-family: 'Open Sans', sans-serif;
  background-color: #ffffff;
  overflow-x: hidden;
  font-size: 14px;
  width: '100%';
}

a {
  text-decoration: none;
}
