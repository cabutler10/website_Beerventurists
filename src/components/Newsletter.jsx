import React, { useState } from 'react'
import addToMailchimp from 'gatsby-plugin-mailchimp'
import { useTranslation } from 'react-i18next'
import { makeStyles } from '@material-ui/styles'
import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogTitle from '@material-ui/core/DialogTitle'
import TextField from '@material-ui/core/TextField'
import Snackbar from '@material-ui/core/Snackbar'
import FormControlLabel from '@material-ui/core/FormControlLabel'
import Checkbox from '@material-ui/core/Checkbox'
import IconButton from '@material-ui/core/IconButton'
import CloseIcon from '@material-ui/icons/Close'

const useStyles = makeStyles(theme => ({
  containerNewsletter: {
    marginTop: 15,
  },
  form: {
    display: 'flex',
    flexDirection: 'column',
  },
  textField: {
    marginBottom: 15,
  },
  button: {
    padding: '9px 27px',
    fontSize: 14,
    [theme.breakpoints.up('md')]: {
      padding: '16px 48px',
      fontSize: 18,
    },
  },
  buttonSm: {
    fontSize: 12,
  },
  snackbarError: {
    background: theme.status.red,
  },
  snackbarSuccess: {
    background: theme.status.green,
  },
  dialogContent: {
    paddingBottom: 0,
  },
  legal: {
    fontSize: 11,
    marginTop: 15,
  },
  legalContainer: {
    display: 'flex',
    alignItems: 'center',
  },
  img: {
    marginRight: 15,
    marginLeft: 15,
    height: 60,
  },
  dialogActions: {
    marginLeft: 30,
    marginRight: 30,
    marginBottom: 30,
  },
  close: {
    float: 'right',
  },
  actionButton: {
    margin: 0,
  },
  error: {
    color: theme.palette.error.main,
  },
}))

function Newsletter({ variant }) {
  const classes = useStyles()
  const { t } = useTranslation()
  const [email, setEmail] = useState('')
  const [name, setName] = useState('')
  const [isEmailValid, setEmailValid] = useState(null)
  const [isNameValid, setNameValid] = useState(null)
  const [isCheckboxOpen, setCheckbox] = useState(false)
  const [isDialogOpen, setDialog] = useState(false)
  const [isSnackbarOpen, setSnackbar] = useState(false)
  const [snackbarMessage, setSnackbarMessage] = useState(false)
  const [error, setError] = useState(false)
  const pattern = /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/

  const handleSubmit = async e => {
    if (isCheckboxOpen && isEmailValid && isNameValid) {
      const result = await addToMailchimp(email, {
        FNAME: name,
        gdpr_26529: true,
      })
      setDialog(false)
      if (result.result === 'error') {
        setSnackbar(true)
        setSnackbarMessage(
          result.msg.includes('<a') ? result.msg.split('<a')[0] : result.msg
        )
      } else {
        setSnackbar(true)
        setError(false)
        setEmailValid(true)
        setNameValid(true)
        setSnackbarMessage('success')
      }
    }
    if (!isCheckboxOpen) {
      setError(true)
    }
    if (!pattern.test(email)) {
      setEmailValid(false)
    }
    if (name.length <= 500) {
      setNameValid(false)
    }
  }

  const handleSnackbarClose = (e, reason) => {
    if (reason === 'clickaway') {
      return
    }
    setSnackbar(false)
  }

  const handleChange = name => event => {
    if (name === 'email') {
      setEmail(event.target.value)
      if (!pattern.test(event.target.value)) {
        setEmailValid(false)
      } else {
        setEmailValid(true)
      }
    } else if (name === 'name') {
      if (event.target.value.length > 500) {
        setNameValid(false)
      } else {
        setNameValid(true)
        setName(event.target.value)
      }
    } else if ((name = 'checkbox')) {
      setCheckbox(event.target.checked)
    }
  }

  return (
    <div className={classes.containerNewsletter}>
      <Snackbar
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'left',
        }}
        open={isSnackbarOpen}
        autoHideDuration={5000}
        onClose={handleSnackbarClose}
        ContentProps={{
          'aria-describedby': 'message-id',
          className:
            snackbarMessage === 'success'
              ? classes.snackbarSuccess
              : classes.snackbarError,
        }}
        message={
          snackbarMessage === 'success' ? (
            <span id="message-id">{t('newsletter.success')}</span>
          ) : (
            <span id="message-id">{snackbarMessage}</span>
          )
        }
        action={
          <IconButton
            key="closeNewsletter"
            aria-label={t('common.close')}
            color="inherit"
            onClick={handleSnackbarClose}
          >
            <CloseIcon />
          </IconButton>
        }
      />

      <Button
        variant="contained"
        color="primary"
        className={variant === 'small' ? classes.buttonSm : classes.button}
        fullWidth={variant === 'small' ? null : true}
        onClick={() => setDialog(true)}
      >
        {t(variant === 'small' ? 'common.newsletter' : 'common.signUp')}
      </Button>
      <Dialog open={isDialogOpen} onClose={() => setDialog(false)}>
        <DialogTitle id="dialog-title">
          {t('common.newsletter')}
          <IconButton
            aria-label={t('common.close')}
            className={classes.close}
            color="inherit"
            onClick={() => setDialog(false)}
          >
            <CloseIcon />
          </IconButton>
        </DialogTitle>
        <DialogContent className={classes.dialogContent}>
          <form className={classes.form}>
            <TextField
              required
              id="email"
              type="text"
              label="email"
              error={isEmailValid === null ? null : !isEmailValid}
              placeholder={t('common.email')}
              value={email}
              onChange={handleChange('email')}
              className={classes.textField}
              margin="normal"
              variant="outlined"
            />
            <TextField
              required
              id="name"
              type="text"
              label="name"
              error={isNameValid === null ? null : !isNameValid}
              placeholder={t('common.name')}
              value={name}
              onChange={handleChange('name')}
              className={classes.textField}
              margin="normal"
              variant="outlined"
            />
          </form>
          <div className={classes.legalContainer}>
            <FormControlLabel
              classes={{
                label: error && !isCheckboxOpen ? classes.error : null,
              }}
              control={
                <Checkbox
                  checked={isCheckboxOpen}
                  onChange={handleChange('checkbox')}
                  className={error ? classes.error : null}
                  value="consent"
                  color="primary"
                />
              }
              label={t('newsletter.consent')}
            />
          </div>
          <Typography className={classes.legal}>
            {t('newsletter.legal1')}
            <a href="https://mailchimp.com/legal/">{t('newsletter.legal2')}</a>
          </Typography>
        </DialogContent>
        <DialogActions className={classes.dialogActions}>
          <Button
            onClick={() => setDialog(false)}
            className={classes.actionButton}
          >
            {t('common.cancel')}
          </Button>
          <Button
            data-testid="submit"
            onClick={handleSubmit}
            className={classes.actionButton}
            color="primary"
            variant="contained"
            autoFocus
            type="submit"
          >
            {t('common.subscribe')}
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  )
}

export default Newsletter
