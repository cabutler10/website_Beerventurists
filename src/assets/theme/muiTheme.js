import { createMuiTheme } from '@material-ui/core/styles'
import 'typeface-open-sans'
import 'typeface-permanent-marker'
import './app.css'

const status = {
  white: '#ffffff',
  black: '#040001',
  greyLt: '#F7F5F5',
  greyMed: '#656565',
  grey: '#c5c6c7',
  greyDk: '#404040',
  green: '#00942b',
  yellow: '#fce566',
  red: '#FF0012',
  blue: '#015b99',
  blueLt: '#0172c0',
  blueDk: '#014472',
  orange: '#c56113',
  orangeDk: '#c56113',
  orangeLt: '#ebab2f',
  tan: '#FFFBEF',
}

const theme = createMuiTheme({
  palette: {
    type: 'light',
    primary: {
      light: status.blueLt,
      main: status.blue,
      dark: status.blueDk,
    },
    secondary: {
      light: status.orangeLt,
      main: status.orange,
      dark: status.orangeDk,
    },
    error: { main: status.red },
  },
  status,
  typography: {
    useNextVariants: true,
    fontFamily: "'Open Sans', sans-serif",
    h5: {
      letterSpacing: 3,
    },
    h2: {
      fontFamily: "'Permanent Marker', cursive",
    },
    body1: {
      letterSpacing: 1,
      lineHeight: 1.8,
      //TODO is this an error in MUI? not sure why its default is 16 when MUI default is 14
      fontSize: 14,
    },
    body2: {
      letterSpacing: 1,
      fontSize: 14,
    },
    button: {
      letterSpacing: 3,
    },
  },
  overrides: {
    MuiTypography: {
      colorPrimary: {
        color: status.black,
      },
      colorSecondary: {
        color: status.orangeLt,
      },
      colorTextSecondary: {
        color: status.white,
      },
    },
    MuiButton: {
      root: {
        padding: '9px 27px',
        borderRadius: 0,
      },
      contained: {
        boxShadow: 'none',
      },
    },
  },
})

export default theme
