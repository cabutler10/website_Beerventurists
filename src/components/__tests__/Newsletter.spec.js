import React from 'react'
import { render } from 'react-testing-library'
import Newsletter from '../Newsletter'
import { ThemeProvider } from '@material-ui/styles'
import theme from '../../assets/theme/muiTheme'

describe('Newsletter', () => {
  const mockProps = {
    t: () => '',
  }
  it('renders correctly', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Newsletter {...mockProps} />
      </ThemeProvider>
    )
    expect(component).toMatchSnapshot()
  })
})
