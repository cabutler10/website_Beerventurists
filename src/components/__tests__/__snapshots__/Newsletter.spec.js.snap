// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newsletter renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Hook-containerNewsletter-1aq479i"
      >
        <button
          class="MuiButtonBase-root-34 MuiButton-root-8 MuiButton-contained-19 MuiButton-containedPrimary-20 MuiButton-raised-22 MuiButton-raisedPrimary-23 MuiButton-fullWidth-33 Hook-button-1j9lk2z"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-9"
          >
            newsletter.signUp
          </span>
          <span
            class="MuiTouchRipple-root-53"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Hook-containerNewsletter-1aq479i"
    >
      <button
        class="MuiButtonBase-root-34 MuiButton-root-8 MuiButton-contained-19 MuiButton-containedPrimary-20 MuiButton-raised-22 MuiButton-raisedPrimary-23 MuiButton-fullWidth-33 Hook-button-1j9lk2z"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-9"
        >
          newsletter.signUp
        </span>
        <span
          class="MuiTouchRipple-root-53"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
